From 7d89214747025d94a5686a761ce1f8ceef223100 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Wed, 26 Sep 2018 15:14:58 +0200
Subject: [PATCH 1/1] dhcp: dhclient: fix memory leak

Fixes: c263f5355cbd86da75aaa161987fee8b59cb8c8b
(cherry picked from commit 0ba0f52cb74375a4a3b2a2f893f7b617510d80fd)
(cherry picked from commit 147081bd72c896b944059ac1538cf599b3b95632)
---
 src/dhcp/nm-dhcp-dhclient.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/dhcp/nm-dhcp-dhclient.c b/src/dhcp/nm-dhcp-dhclient.c
index 4a9361781..33c267123 100644
--- a/src/dhcp/nm-dhcp-dhclient.c
+++ b/src/dhcp/nm-dhcp-dhclient.c
@@ -142,6 +142,7 @@ get_dhclient_leasefile (int addr_family,
 	                        iface);
 
 	if (g_file_test (path, G_FILE_TEST_EXISTS)) {
+		g_free (rundir_path);
 		NM_SET_OUT (out_preferred_path, g_strdup (path));
 		return path;
 	}
-- 
2.17.1

